### ISSUE TITLE
Codespaces: Terminal: MesloLGS NF's (Nerd Font patched) glyphs and symbols not rendered properly

### ISSUE THREAD

**Issue by benz0li** [microsoft/vscode#231725]
Does this issue occur when all extensions are disabled?: Not relevant

- VS Code Version: 1.94.2, 384ff7382de624fb94dbaf6da11977bba1ecd427, x64
- OS Version: Debian 12 (bookworm)

Steps to Reproduce:

1. Open https://github.com/b-data/data-science-devcontainers
2. Codespaces > New with options...
    - Dev container configuration: Python base
3. Install font Meslo LGS NF

    ```bash
    mkdir -p ${HOME}/.local/share/fonts/truetype/meslo
    curl -sSL https://github.com/romkatv/powerlevel10k-media/raw/master/MesloLGS%20NF%20Regular.ttf -o "${HOME}/.local/share/fonts/truetype/meslo/MesloLGS NF Regular.ttf"
    curl -sSL https://github.com/romkatv/powerlevel10k-media/raw/master/MesloLGS%20NF%20Bold.ttf -o "${HOME}/.local/share/fonts/truetype/meslo/MesloLGS NF Bold.ttf"
    curl -sSL https://github.com/romkatv/powerlevel10k-media/raw/master/MesloLGS%20NF%20Italic.ttf -o "${HOME}/.local/share/fonts/truetype/meslo/MesloLGS NF Italic.ttf"
    curl -sSL https://github.com/romkatv/powerlevel10k-media/raw/master/MesloLGS%20NF%20Bold%20Italic.ttf -o "${HOME}/.local/share/fonts/truetype/meslo/MesloLGS NF Bold Italic.ttf"
    fc-cache -fv
    ```
4. Install Zsh theme Powerlevel10k

    ```bash
    git clone --depth=1 https://github.com/romkatv/powerlevel10k.git ${HOME}/.oh-my-zsh/custom/themes/powerlevel10k
    ```
5. Change Zsh theme to Powerlevel10k

    ```bash
    sed -i 's/ZSH_THEME="devcontainers"/ZSH_THEME="powerlevel10k\/powerlevel10k"/g' ${HOME}/.zshrc
    ```
6. Change Settings (User) > Features > Terminal > Integrated: Font Family: `MesloLGS NF`
7. Open a new Terminal

---

Safari v18.0.1 (20619.1.26.31.7); Terminal › Integrated: Gpu Acceleration: auto

Does this look like a diamond (rotated square)? Yes

![Image](https://github.com/user-attachments/assets/a870c233-76ad-40ef-a602-505cfbdb1ada)

Does this look like a lock? No

![Image](https://github.com/user-attachments/assets/5b1294f8-7d76-46d8-921d-9cf2992bf5a8)

Cross reference:

* https://github.com/coder/code-server/pull/7026#issuecomment-2423630849

**Comment by benz0li** [comment]
Safari v18.0.1 (20619.1.26.31.7); Terminal › Integrated: Gpu Acceleration: off

Does this look like a diamond (rotated square)? No

![Image](https://github.com/user-attachments/assets/ef96244f-a0a8-47b4-8f63-80342926dc7c)

**Comment by benz0li** [comment]
Firefox 131.0.3; Terminal › Integrated: Gpu Acceleration: auto

Does this look like a diamond (rotated square)? Yes

![Image](https://github.com/user-attachments/assets/71a2a8c7-c357-4c0d-a259-40757d7b009d)

Does this look like a lock? No

![Image](https://github.com/user-attachments/assets/bd3a92a4-9cad-412c-826f-0bfa0b14da41)

**Comment by benz0li** [comment]
Firefox 131.0.3; Terminal › Integrated: Gpu Acceleration: off

Does this look like a diamond (rotated square)? No

![Image](https://github.com/user-attachments/assets/e21b4132-59bf-497b-a6a1-72a7653bbad1)

Ping @Tyriar

**Comment by benz0li** [comment]
Interestingly, Firefox displays the Terminal with MesloLGS NF correctly at https://coder.jupyter.b-data.ch (code-server 4.94.2-rc.1 aka Code 1.94.2).

**Comment by benz0li** [comment]
The Terminal with MesloLGS NF worked fine for both Safari and Firefox up to code-server version 4.93.1 – aka Code 1.93.1.

This version is currently deployed at https://demo.jupyter.b-data.ch.

**Comment by Sovanda-Mith** [comment]
Version: 1.95.0
Os: Linux x64 6.8.0-47-generic snap 

The issue still persists, my integrated terminal that uses the font family MesloLGS NF cannot rendered glyphs and symbols properly, what could be causing this problem and will there be a fix for this soon?

**Comment by benz0li** [comment]
The problem is resolved for code-server v4.95.1-rc.2 – aka Code 1.95.1.

The problem still persists on Codespaces.

No problem with VS Code.

**Comment by benz0li** [comment]
As I need to patch `code-server` to make this[^1] work, I do not think this works with Codespaces *as is*.

[^1]: Using a patched Nerd Font as Terminal Font Family.
